### ISSUE TITLE
[FF] Include auto-completion to vomnibar

### ISSUE THREAD

**Issue by pcuinty** [philc/vimium#2848]
Is it possible to include auto-completion to the vomnibar similar to the address bar of firefox? Right now I always have to use <ctrl-j> to select to the first entry in the list and then hit return.

**Comment by smblott-github** [comment]
That is the behaviour you get if you use [search completion](https://github.com/philc/vimium/wiki/Search-Completion#available-completion-engines).

For example, add a custom search engine...

    g: https://www.google.com/search?q=%s Google

And add a key binding...

   map s LinkHints.activateMode keyword=g

Then `s` will give you search completion with the auto selection you want.

Auto selecting the first element is a bad UX in the general case because, if you want to search, then you have to `<Shift-Tab>` back into the input (which is awkward).

**Comment by pcuinty** [comment]
Your described method doesn't work for bookmarks or browsing history.

I agree auto selecting the first element is not good. What I meant is using the fist element as an auto complete appendix prpposal like so:
![screenshot_2017-12-01_15-59-07-cropped](https://user-images.githubusercontent.com/14926781/33488496-13982838-d6b1-11e7-999b-851891caaa96.jpg)

I can continue typing or accept the auto completion proposal with return. no need for <shift-tab>

**Comment by smblott-github** [comment]
And how to a search for just "thegu", if that's what I want?

I hate that UX.  You have to keep saying "no, that's not what I want".  It's better to have `<Tab>` mean "yes, that is what I want".

In Vimium, the first element is selected in all of the cases where it makes sense to do so (basically, bookmarks, tabs and search completion -- kind of).

**Comment by Troyciv** [comment]
I see what you mean, you got a valid point. I usually just hit space before I type in the search term I'm looking for. But I guess it ends up in a matter of personal preference. Either you hit tab to select autocompletion or you it space to prevent autocompletion. Doesn't make much difference time-wise.

**Comment by JJPandari** [comment]
@smblott-github You're right about vimium's current UX being better than browsers' address bars' *alway-auto-complete*. So how about add a key binding when vomni bar is active, which completes the vomni bar with the selected item's full address?  

My use case is as follows: I bring up vomni bar to search for some javascript document, I type `moz array`, and I get `https://developer.mozilla.org/.../Global_Objects/Array/from` and `https://developer.mozilla.org/.../Global_Objects/Array/slice` in the suggestion list, everything is good. Except sometimes I may want to look at the doc for `splice`, which is not in the suggestions, but can be easily reached if I can modify the full address of `https://developer.mozilla.org/.../Global_Objects/Array/slice`. Right now I have to go the the `slice` page first before changing address to `splice`.

**Comment by smblott-github** [comment]
I think that's #2914, @JJPandari.

**Comment by ruocco** [comment]
> And how to a search for just "thegu", if that's what I want?
> 
> I hate that UX. You have to keep saying "no, that's not what I want". It's better to have `<Tab>` mean "yes, that is what I want".
> 
> In Vimium, the first element is selected in all of the cases where it makes sense to do so (basically, bookmarks, tabs and search completion -- kind of).

It would be good to have the choice don't you think? Something like:

- [ ] Automatically select the first element on the vomnibar on send

Disabled by default, but allowed nonetheless for people who find it useful.
